.github/workflows/wheel_linux_x86_64.yml

```
@@ -42,45 +43,161 @@ jobs:
+                          -G Ninja
+          cmake --build ./Build --verbose
+          cmake --install ./Build
+          cd -
+
+      - name: Build Kokkos kernels library
+        if: steps.kokkos-cache.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p /root/Kokkos_install/${{ matrix.exec_model }}
+          cd kokkos-kernels
+          cmake -BBuild . -DCMAKE_INSTALL_PREFIX=/root/Kokkos_install/${{ matrix.exec_model }} \
+                          -DKokkos_ENABLE_${{ matrix.exec_model }}=ON \
+                          -DCMAKE_CXX_COMPILER=${{ env.COMPILER }} \
+                          -DCMAKE_CXX_STANDARD=20 \
+                          -DCMAKE_PREFIX_PATH=/root/Kokkos_install/${{ matrix.exec_model }} \
+                          -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+                          -G Ninja
+          cmake --build ./Build --verbose
+          cmake --install ./Build
+
+  linux-wheels-x86-64:
+    needs: [linux-set-matrix-x86_64, build_dependencies]
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
         arch: [x86_64]
-        cibw_build: ${{fromJson(needs.linux-set-matrix-x86_64.outputs.matrix)}}
+        cibw_build: ${{ fromJson(needs.linux-set-matrix-x86_64.outputs.matrix) }}
+        exec_model: ${{ fromJson(needs.linux-set-matrix-x86_64.outputs.exec_model) }}
+        kokkos_version: ${{ fromJson(needs.linux-set-matrix-x86_64.outputs.kokkos_version) }}
+        container_img: ["quay.io/pypa/manylinux2014_x86_64"]
 
     name: ${{ matrix.os }} (Python ${{ fromJson('{ "cp37-*":"3.7","cp38-*":"3.8","cp39-*":"3.9","cp310-*":"3.10" }')[matrix.cibw_build] }})
     runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container_img }}
 
     steps:
       - name: Cancel Previous Runs
         uses: styfle/cancel-workflow-action@0.4.1
         with:
           access_token: ${{ github.token }}
 
-      - uses: actions/checkout@v2
-
-      - uses: actions/setup-python@v2
-        name: Install Python
+      - name: Restoring cached dependencies
+        id: kokkos-cache
+        uses: actions/cache@v3
         with:
-          python-version: '3.7'
+          path: /root/Kokkos_install/${{ matrix.exec_model }}
+          key: ${{ matrix.container_img }}-kokkos${{ matrix.kokkos_version }}-${{ matrix.exec_model }}
+
+      - uses: actions/checkout@v3
+
+      - name: Copy cached libraries
+        run: |
+          mkdir Kokkos
+          cp -rf /root/Kokkos_install/${{ matrix.exec_model }}/* Kokkos/
+
+      - name: Install dependencies (CentOS)
+        if: ${{ (matrix.container_img == 'quay.io/pypa/manylinux2014_x86_64') }}
+        run: |
+          # Reduce wait time for repos not responding
+          cat /etc/yum.conf | sed "s/\[main\]/\[main\]\ntimeout=5/g" > /etc/yum.conf
+          yum update -y && yum install -y docker
 
       - name: Install cibuildwheel
-        run: python -m pip install cibuildwheel==2.3.0
+        run: python3.7 -m pip install cibuildwheel==2.8.1
```

```suggestion
        run: python3 -m pip install cibuildwheel==2.8.1
```

We probably don't need to pin to 3.7

---

In fact, we need, since we are building inside a container.